File: zig/src/render/codeFenceRenderer.zig
const std = @import("std");
const MarkdownNode = @import("../types/MarkdownNode.zig").MarkdownNode;
const RendererState = @import("../types/RendererState.zig").RendererState;
const Renderer = @import("../types/Renderer.zig").Renderer;
const render = @import("codeBlockRenderer.zig").render;

pub const codeFenceRenderer = Renderer{
    .name = "code_fence",
    .render = render.render,
};

File: dotnet/Allmark/Render/CodeFenceRenderer.cs
namespace Allmark.Render;

using Allmark.Types;

public static class CodeFenceRenderer
{
    public static OutputRenderer Create()
    {
        return new OutputRenderer
        {
            Name = "code_fence",
            Render = Render,
        };
    }

    public static void Render(MarkdownNode node, RendererState state, bool? decode = true)
    {
        RenderUtils.StartNewLine(node, state);
        var lang = "";
        if (!string.IsNullOrEmpty(node.Info))
        {
            var trimmed = node.Info.Trim().Split(' ')[0];
            if (!string.IsNullOrEmpty(trimmed))
            {
                lang = $" class=\"language-{Utils.EscapeHtml(trimmed)}\"";
            }
        }
        state.Output.Append($"<pre><code{lang}>");
        RenderChildren.Execute(node, state, false);
        state.Output.Append("</code></pre>");
        RenderUtils.EndNewLine(node, state);
    }
}
